crates/ruma-client-api/src/sync/sync_events/v3.rs

```diff
@@ -349,7 +349,6 @@ pub struct Timeline {
     pub prev_batch: Option<String>,
 
     /// A list of events.
-    #[serde(default, skip_serializing_if = "Vec::is_empty")]
```

Comment:
This doesn't make sense to me (while the rest of the PR does, and I'm a bit shocked we had a bug like that for so long). We definitely can't remove the `default` attribute, and I also don't see why we would no longer skip serializing the  `events` part of the timeline if it's empty. Is there a client that breaks if there is a timeline object but it doesn't include this field?

Reply:
Ah, good catch, I didn't notice the `default` attribute when I removed this line. The reason to remove `skip_serializing_if` is that the field is marked as required in [the spec](https://spec.matrix.org/v1.10/client-server-api/#get_matrixclientv3sync). The fact that we were skipping it didn't really matter before, because we were skipping the entire `Timeline` struct.

Reply:
Right, then we should actually not have either serde annotation.